File: expohub-backend/app/models/booth.py
"""
展位表（Booth）

对应前端 API: expo-hub-uniapp/src/api/booth.ts

前端 Booth 接口字段：
  id, exhibition_id, booth_number, exhibitor_id?, exhibitor_name?,
  company_name?, size?, location_area?, price?, status,
  description?, created_at?, updated_at?
"""

from datetime import datetime
from typing import Optional

from sqlalchemy import String, DateTime, func, Text, Integer, Float, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Booth(Base):
    """展位表"""
    __tablename__ = "booths"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)

    # ---- 关联 ----
    exhibition_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("exhibitions.id"), nullable=False, index=True,
        comment="所属展会 ID"
    )
    exhibitor_id: Mapped[Optional[int]] = mapped_column(
        Integer, ForeignKey("users.id"), nullable=True, index=True,
        comment="展商用户 ID（空表示未分配）"
    )

    # ---- 展位标识 ----
    booth_number: Mapped[str] = mapped_column(
        String(50), nullable=False, comment="展位编号，如 A-001"
    )

    # ---- 冗余展示信息（与前端 Booth 接口对齐） ----
    exhibitor_name: Mapped[Optional[str]] = mapped_column(
        String(100), nullable=True, comment="展商名称（冗余）"
    )
    company_name: Mapped[Optional[str]] = mapped_column(
        String(200), nullable=True, comment="公司名称（冗余）"
    )

    # ---- 展位属性（与前端接口对齐） ----
    size: Mapped[Optional[str]] = mapped_column(
        String(50), nullable=True, comment="展位尺寸，如 3x3m"
    )
    location_area: Mapped[Optional[str]] = mapped_column(
        String(100), nullable=True, comment="展区位置，如 A区/主厅"
    )
    price: Mapped[Optional[float]] = mapped_column(
        Float, nullable=True, comment="展位价格"
    )
    description: Mapped[Optional[str]] = mapped_column(
        Text, nullable=True, comment="展位描述"
    )

    # ---- 状态 ----
    status: Mapped[str] = mapped_column(
        String(20), nullable=False, default="available", index=True,
        comment="状态：available/reserved/occupied/maintenance"
    )

    # ---- 时间戳 ----
    created_at: Mapped[Optional[datetime]] = mapped_column(
        DateTime, server_default=func.now(), comment="创建时间"
    )
    updated_at: Mapped[Optional[datetime]] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now(),
        comment="更新时间"
    )

    def __repr__(self) -> str:
        return f"<Booth(id={self.id}, number='{self.booth_number}')>"

File: expo-hub-uniapp/src/api/booth.ts
import http from './index'

export interface Booth {
  id: number
  exhibition_id: number
  booth_number: string
  exhibitor_id?: number
  exhibitor_name?: string
  company_name?: string
  size?: string
  location_area?: string
  price?: number
  status: string
  description?: string
  created_at?: string
  updated_at?: string
}

export const boothApi = {
  getList(params?: { exhibition_id?: number; page?: number; page_size?: number; status?: string }) {
    return http.get('/booths', { params }).then((r) => r.data)
  },

  getById(id: number | string) {
    return http.get(`/booths/${id}`).then((r) => r.data)
  },

  create(data: Partial<Booth>) {
    return http.post('/booths', data).then((r) => r.data)
  },

  update(id: number | string, data: Partial<Booth>) {
    return http.put(`/booths/${id}`, data).then((r) => r.data)
  },

  delete(id: number | string) {
    return http.delete(`/booths/${id}`).then((r) => r.data)
  },

  apply(boothId: number | string) {
    return http.post(`/booths/${boothId}/apply`).then((r) => r.data)
  },

  getMyBooths(params?: { page?: number; page_size?: number }) {
    return http.get('/booths/my', { params }).then((r) => r.data)
  },
}
